app/assets/src/components/views/phylo_tree/PhyloTreeVis.jsx

```diff
@@ -33,26 +46,91 @@ class PhyloTreeVis extends React.Component {
       // Name for the legend when the attribute is missing / other
       colorGroupAbsentName: "NCBI References",
       legendX: 880,
-      legendY: 50
+      legendY: 50,
+      tooltipContainer: this.tooltipContainer,
+      onNodeTextClick: this.handleNodeClick,
+      onNodeHover: this.handleNodeHover
     });
     this.treeVis.update();
   }
 
   componentDidUpdate() {
-    this.treeVis.setTree(
-      Tree.fromNewickString(this.props.newick, this.props.nodeData)
-    );
-    this.treeVis.update();
+    if (
+      this.props.newick != this.newick ||
+      this.props.nodeData != this.nodeData
+    ) {
+      this.newick = this.props.newick;
+      this.nodeData = this.props.nodeData;
+      this.treeVis.setTree(
+        Tree.fromNewickString(this.props.newick, this.props.nodeData)
+      );
+      this.treeVis.update();
+    }
+  }
+
+  handleNodeHover(node) {
+    this.setState({ hoveredNode: node });
+  }
+
+  handleNodeClick(node) {
+    if (node.data.accession) {
+      let url = `https://www.ncbi.nlm.nih.gov/nuccore/${node.data.accession}`;
+      window.open(url, "_blank");
```

Comment:
Add rel="noopener noreferrer" to all external links https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/

Reply:
or https://mathiasbynens.github.io/rel-noopener/